src/common/gamedata.rs

```diff
@@ -568,13 +569,184 @@ impl GameData {
+
+                let physis::exd::ColumnData::Int32(aether_current_8) = &single_row.columns[9] else {
+                    panic!("Unexpected type!");
+                };
+
+                let physis::exd::ColumnData::Int32(aether_current_9) = &single_row.columns[10] else {
+                    panic!("Unexpected type!");
+                };
+
+                let physis::exd::ColumnData::Int32(aether_current_10) = &single_row.columns[11] else {
+                    panic!("Unexpected type!");
+                };
+
+                let physis::exd::ColumnData::Int32(aether_current_11) = &single_row.columns[12] else {
+                    panic!("Unexpected type!");
+                };
+
+                let physis::exd::ColumnData::Int32(aether_current_12) = &single_row.columns[13] else {
+                    panic!("Unexpected type!");
+                };
+
+                let physis::exd::ColumnData::Int32(aether_current_13) = &single_row.columns[14] else {
+                    panic!("Unexpected type!");
+                };
+
+                let physis::exd::ColumnData::Int32(aether_current_14) = &single_row.columns[15] else {
+                    panic!("Unexpected type!");
+                };
+
+                if *aether_current_0 == aether_current_id ||
+                *aether_current_1 == aether_current_id ||
+                *aether_current_2 == aether_current_id ||
+                *aether_current_3 == aether_current_id ||
+                *aether_current_4 == aether_current_id ||
+                *aether_current_5 == aether_current_id ||
+                *aether_current_6 == aether_current_id ||
+                *aether_current_7 == aether_current_id ||
+                *aether_current_8 == aether_current_id ||
+                *aether_current_9 == aether_current_id ||
+                *aether_current_10 == aether_current_id ||
+                *aether_current_11 == aether_current_id ||
+                *aether_current_12 == aether_current_id ||
+                *aether_current_13 == aether_current_id ||
+                *aether_current_14 == aether_current_id {
+                    result = Some(&row.row_id);
+                    break 'outer;
+                }
+            }
+        }
+
+        if let Some(item_id) = result {
+            return Some(*item_id);
+        }
+
+        None
+    }
+
+    /// Gets the Aether Currents needed for a zone.
+    pub fn get_aether_currents_from_zone(&mut self, aether_current_comp_flg_set_id: u32) -> Option<Vec<i32>> {
+        let mut aether_currents = Vec::new();
+
+        let aether_current_comp_flg_set_sheet = 
+            AetherCurrentCompFlgSetSheet::read_from(&mut self.resource, Language::None).unwrap();
+
+        let row = aether_current_comp_flg_set_sheet.get_row(aether_current_comp_flg_set_id)?;
+        let aether_currents_from_zone = row.AetherCurrents().map(|x| *x.into_i32().unwrap());
+
+        for aether_current_from_zone in aether_currents_from_zone {
```

Comment:
You can technically simplify this with a `filter` since you're just looking for non-zero values.

Reply:
Done!